code/game/area/Space_Station_13_areas.dm

```diff
@@ -618,6 +618,9 @@ NOTE: there are two lists of areas in the end of this file: centcom and station
 	lighting_colour_bulb = "#ffebc1"
 	sound_environment = SOUND_AREA_WOODFLOOR
 
+/area/crew_quarters/bar/mood_check(mob/living/carbon/human/subject)
+		return HAS_TRAIT(subject, TRAIT_LIGHT_DRINKER) // light drinkers dont like drinks, wow
```

Comment:
```suggestion
/area/crew_quarters/bar/mood_check(mob/living/carbon/human/subject)
		. = ..()
		if (!HAS_TRAIT(subject, TRAIT_LIGHT_DRINKER))
				. = FALSE
```
This should handle the base case actually

Reply:
dont you mean HAS_TRAIT without a !

Reply:
Oh yea

Reply:
Original should have been inverted